### PR TITLE
layers: Header cleanup

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Remove more `#include` headers that are just including the world everywhere

### CODE REVIEW

File: layers/best_practices/best_practices_validation.h [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <string>
 #include <deque>
 #include <chrono>
+#include <set>
```
Comment (artem-lunarg) [inline layers/best_practices/best_practices_validation.h:28]: Does this change remove includes that are already included by some other headers? It looks like `std::set` is used in this file.

Comment (artem-lunarg) [reply layers/best_practices/best_practices_validation.h:28]: In general it's not recommended to rely on include dependencies and include what is used (can depend on compiler), but not sure what clang tool is actually doing.

**Review by artem-lunarg — APPROVED** [review]
LGTM assuming it's CI issue

File: layers/best_practices/best_practices_validation.h (hunk 1 above)
Comment (spencer-lunarg) [reply layers/best_practices/best_practices_validation.h:28]: as you might have found out, the `custom_containers.h` wasn't hiding `<set>`/`<map>` in the `#ifdef USE_ROBIN_HOOD_HASHING` else check

**Comment by spencer-lunarg** [comment]
CI is failing for Video test (unrelated) and tracking it here https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8696

**Pull request merged**